app/src/main/kotlin/com/wire/android/navigation/NavigationUtils.kt

```diff
@@ -34,3 +36,17 @@ internal fun NavController.getCurrentNavigationItem(): NavigationItem? {
     val currentRoute = navBackStackEntry?.destination?.route
     return NavigationItem.fromRoute(currentRoute)
 }
+
+internal fun QualifiedID.mapIntoArgumentString(): String = "$domain@$value"
+internal fun ImageAsset.PrivateAsset.mapIntoArgumentsString(): String = "${conversationId.mapIntoArgumentString()}:$messageId"
+
+fun String.parseIntoQualifiedID(): QualifiedID {
+    val components = split("@")
+    return QualifiedID(components.last(), components.first())
+}
```

Comment:
Same here we do have `toConversationId` that extract the qualifiedId from a string